sherpa/utils/src/_psf.cc

```diff
@@ -273,48 +294,30 @@ static PyObject* tcdPyData_convolve( tcdPyData* self, PyObject* args )
 			  &center) )
     return NULL;
   
-  if( dims_src.get_size() != dims_kern.get_size() ) {
-    std::ostringstream err;
-    err << "input array sizes do not match, "
-	<< "dims_src: " << dims_src.get_size()
-	<< " vs dims_kern: " << dims_kern.get_size();
-    PyErr_SetString( PyExc_TypeError, err.str().c_str() );
+  if( !same_size_arrays( dims_src.get_size(), dims_kern.get_size(),
+                         ", ", "dims_src: ", " vs dims_kern: " ) )
```

Comment:
arguments to same_size_arrays() should not contain formatting (such spaces, colons, commas) those should be handled in the same_size_arrays error string generation. For example, you should be passing in "dims_src" instead of "dims_src: ". (This applies to calls to same_size_array throughout the file.)